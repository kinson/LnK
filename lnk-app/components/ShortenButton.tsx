interface Props {
  disabled: boolean;
  onClick: () => void;
}

export default function ShortenButton({ disabled, onClick }: Props) {
  const baseStyle =
    'bg-purple mr-0.5 px-3 py-2 rounded-tl-md rounded-bl-md text-white';

  const style = disabled
    ? `${baseStyle} bg-opacity-60 text-opacity-70`
    : baseStyle;

  return (
    <button
      disabled={disabled}
      onClick={onClick}
      type="button"
      className={style}
    >
      Shorten
    </button>
  );
}
