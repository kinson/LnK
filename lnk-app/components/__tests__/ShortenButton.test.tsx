import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import ShortenButton from '../ShortenButton';

describe('ShortenButton component tests', () => {
  it('should render the ShortenButton component when it not disabled', () => {
    const component = renderer.create(
      <ShortenButton disabled={false} onClick={jest.fn()} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should render the ShortenButton component when it is disabled', () => {
    const component = renderer.create(
      <ShortenButton disabled onClick={jest.fn()} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should call the onClick prop when clicked', () => {
    const onClickMock = jest.fn();
    const component = render(
      <ShortenButton disabled={false} onClick={onClickMock} />
    );

    fireEvent.click(component.queryAllByText(/Shorten/i)[0]);

    expect(onClickMock).toHaveBeenCalled();
  });
});
