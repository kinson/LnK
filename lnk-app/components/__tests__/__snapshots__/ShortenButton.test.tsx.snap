// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortenButton component tests should render the ShortenButton component when it is disabled 1`] = `
<button
  className="bg-purple mr-0.5 px-3 py-2 rounded-tl-md rounded-bl-md text-white bg-opacity-60 text-opacity-70"
  disabled={true}
  onClick={[MockFunction]}
  type="button"
>
  Shorten
</button>
`;

exports[`ShortenButton component tests should render the ShortenButton component when it not disabled 1`] = `
<button
  className="bg-purple mr-0.5 px-3 py-2 rounded-tl-md rounded-bl-md text-white"
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Shorten
</button>
`;
